src/wallet/mod.rs

```diff
@@ -1894,12 +1875,6 @@ pub(crate) mod test {
         Wallet::db_checksum(&mut db, &desc, KeychainKind::External)
             .expect("db that uses actual checksum should be supported");
 
-        let mut db = MemoryDatabase::new();
-        db.check_descriptor_checksum(KeychainKind::External, checksum_inception)
-            .expect("failed to save checksum inception");
-        Wallet::db_checksum(&mut db, &desc, KeychainKind::External)
-            .expect("db that uses checksum inception should be supported");
-
```

Comment:
I am not entirely sure, but will this create a hard error for database that had "inception bug" checksum?? In v0.24, the logic was if the db had such bad checksum, it will be accepted in the test.. But the bad checksum would still remain in db.. So when someone updates from v0.24 to v0.25 with bad checksum, this should now then cause hard error?

Reply:
I don't think just using the new checksum will cause a hard error. But if a wallet is created with a name and checksum A but then that same wallet name is used and the descriptor now  has checksum B that will be an error.  I expect most wallets use the `Default` name which is the checksum inself, so changing this checksum code will just create a new wallet name and duplicate the cache of wallet data on sync. Either way I think we've been clear enough in the deprecation last release that this is changing.